Update UI components

laurelfulford opened the pull request:
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR is pretty big, but it mostly changes CSS variable values and names. Specifically:
* It updates the colour variable values to match the new component Figma file, and updates the names to match.
* It updates the font size variables, and updates their names to match. 
* It updates the variable names and values of the border-radius & modal widths.
* It updates the variable names throughout the SCSS files.
* It adds a couple more examples to the UI Demo page, for warnings, font sizes, and the branded buttons.

Outside of that, it also:
* Updates the styles of the radio button, checkbox, and text inputs to match the components (changes include adjusting the border radius, padding, and focus styles.
* Builds out the WooCommerce overrides styles, to make sure the checkout is picking up the new input and font sizing styles.

In the non-SCSS, non-demo files, the PR also updates a couple CSS classes in the reader-activation success message and the reader activation login form.

Still to do:
* Set up the different button size styles.
* Updating the ‘checkbox + label + container’ element.
* Update the label element to incorporate the error styles.

See: 1205234045751551-as-1206899964006422

### How to test the changes in this Pull Request:

1. Start on a test site running RAS.
2. Add a donate block to a page.
3. Create a variable product, and add a checkout button block for it to the same page. 
4. Apply the PR and run `npm run build`. 
5. While logged into a test site, append `?ui-demo` to any page’s URL; review that against the Figma file for font sizes, padding, spacing. 
6. Focus on inputs, and click radio buttons. 
7. Click the ‘Open Modal’ button at the bottom and review its appearance. 
8. View the site in “incognito mode”/as a reader. 
9. Run through the Donate process, and spot check the appearance of the modals.
10. Run through the checkout process using the Checkout button, adn spot check the appearance of the variable modal. 
11. Click the Sign In button in the header, and spot-check the appearance of the modal.
12. Test the 'prefers contrast' colours. In Chrome, you can do this by opening the Developer tools, enabling the Rendering tab, and finding the "Emulate CSS media feature prefers-contrast". Change this to prefers-contrast: more:

![image](https://github.com/Automattic/newspack-plugin/assets/177561/ccb1c1b0-dee8-4023-8a86-90dfa91a2fa2)



### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

<!-- Mark completed items with an [x] -->

laurelfulford committed:
feat: update component colors, input styles

laurelfulford committed:
feat: adjust focus styles

laurelfulford committed:
fix: cleaning up some formatting

laurelfulford committed:
fix: correct checkbox, radio focus style

laurelfulford committed:
feat: update border radiuses, some font sizes

laurelfulford committed:
feat: replace values in modal.scss with variables

thomasguillot commented:
We need to differentiate Primary buttons `neutral-90` and Branded buttons `--newspack-theme-color-primary` (or primary-60 in Figma).

It's irrelevant for RAS-ACC but My Account will use both.

thomasguillot committed:
feat: add clamp to font size

thomasguillot commented:
There's something confusing going on. The CSS variables for the font sizes aren't following the Figma naming convention.

E.g.
`--newspack-ui-font-size-large` -> `--newspack-ui-font-size-m`
`--newspack-ui-font-size-extra-small` -> `--newspack-ui-font-size-2xs`

thomasguillot committed:
feat: add elevation mixin

thomasguillot commented:
@laurelfulford I have added a mixin for the Elevation (1, 2, and 3). Feel free to move things around.

laurelfulford committed:
feat: standardize variable names, and update font sizes

laurelfulford commented:
Thanks @thomasguillot! I'm still poking at this, but wanted to drop a quick update here: 

> We need to differentiate Primary buttons neutral-90 and Branded buttons --newspack-theme-color-primary (or primary-60 in Figma).

I've added a second CSS class so we can specifically assign Branded or Primary button styles. Right now, the 'primary' button colour switches to the 'branded' colour when it's used in the modals, based off of a previous discussion. If this is no longer right, just let me know!

There's also some button sizing stuff in the components file that this PR doesn't address (since they're not used in RAS), but that are on my radar 🙂 

> There's something confusing going on. The CSS variables for the font sizes aren't following the Figma naming convention.

This should be fixed, and the variables should now include the full range from 2xs to 6xl. I've also updated some other variable names to follow this pattern (to move from `small` > `s`, `extra-small` > `xs`, but need to do some more checking to make sure I have them all, and to make sure the font sizes are accurately applied.

thomasguillot commented:
> I've added a second CSS class so we can specifically assign Branded or Primary button styles. Right now, the 'primary' button colour switches to the 'branded' colour when it's used in the modals, based off of a previous discussion. If this is no longer right, just let me know!

The modals in My Account will use Primary and not Branded.

thomasguillot commented on assets/newspack-ui/scss/elements/_tables.scss:
```diff
@@ -1,15 +1,15 @@
-		font-size: var( --newspack-ui-font-size-small ); // Prevent relative font size
-		line-height: var( --newspack-ui-line-height-large );
+		font-size: var( --newspack-ui-font-size-xs ); // Prevent relative font size
+		line-height: var( --newspack-ui-line-height-l );
```
The line-height should be `xs`

laurelfulford committed:
feat: tweaking styles that affect the checkout modal

laurelfulford committed:
feat: update button styles, font size, Woo checkout tweaks

laurelfulford committed:
feat: adjust font sizes, line heights, input styles

laurelfulford committed:
feat: add comments for the font sizes

laurelfulford committed:
fix: update comment

thomasguillot committed:
feat: update focus styles

thomasguillot committed:
feat: update buttons and add destructive

thomasguillot committed:
feat: add outline class to button

thomasguillot committed:
fix: wrong classes

thomasguillot committed:
feat: add Button icon

thomasguillot committed:
fix: modal header stickyness

thomasguillot committed:
fix(demo): modal class width

laurelfulford committed:
fix: add baseline line height

laurelfulford committed:
fix: add blanket fixed layout to tables for cell width consistency

laurelfulford committed:
feat: update icon button, update button classes, fix helper text

laurelfulford commented:
I added a PR on the Newspack Blocks side just to update the button class name in a couple spots (`--tertiary` -> `--ghost`): https://github.com/Automattic/newspack-blocks/pull/1713

laurelfulford committed:
feat: merge in updates from epic/ras-acc

adekbadek commented on assets/newspack-ui/scss/_modals.scss:
```diff
@@ -127,61 +126,49 @@
 		&__close {
-			margin: -2px -5px 0 0; // Offsets a padding in the SVG.
-			padding: 0 0 0 8px;
-			background: transparent;
```
The modal checkout close button now inherits button styles, which I don't this is intended:

<img width="792" alt="image" src="https://github.com/Automattic/newspack-plugin/assets/7383192/3d9ec178-b72d-4fe8-a285-17fa0d2ee665">

laurelfulford replied:
Oooo, thanks @adekbadek! This should now be fixed in the Blocks version of the PR, https://github.com/Automattic/newspack-blocks/pull/1713 -- I'll leave it unmerged in case anything still seems off!

laurelfulford commented:
Thanks @adekbadek!
